noir_stdlib/src/hash/poseidon/bn254.nr

```diff
@@ -0,0 +1,107 @@
+// Instantiations of Poseidon constants, permutations and sponge for prime field of the same order as BN254
+mod perm;
+mod consts;
+
+use crate::hash::poseidon::PoseidonConfig;
+use crate::array;
+use crate::pow_32;
+use crate::hash::poseidon::apply_matrix;
+
+// Optimised permutation for this particular field
```

Comment:
Could you add a comment saying that the optimisation is based on the fact that rf and rp are hardcoded in the function instead of coming from function arguments. Once noir can use function arguments in loop range, we should apply this optimisation to the main permutation.